bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcareBridgeHandler.java

```diff
@@ -0,0 +1,239 @@
+import org.openhab.binding.surepetcare.internal.SurePetcareAPIHelper;
+import org.openhab.binding.surepetcare.internal.discovery.SurePetcareDiscoveryService;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcareBridgeConfiguration;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcareDevice;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcareHousehold;
+import org.openhab.binding.surepetcare.internal.dto.SurePetcarePet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SurePetcareBridgeHandler} is responsible for handling the bridge things created to use the Sure Petcare
+ * API. This way, the user credentials may be entered only once.
+ *
+ * It also spawns 2 background polling threads to update the more static data (topology) and the pet locations at
+ * different time intervals.
+ *
+ * @author Rene Scherer - Initial Contribution
+ */
+@NonNullByDefault
+public class SurePetcareBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SurePetcareBridgeHandler.class);
+
+    private final SurePetcareAPIHelper petcareAPI;
+    private @Nullable ScheduledFuture<?> topologyPollingJob = null;
+    private @Nullable ScheduledFuture<?> petStatusPollingJob = null;
+
+    public SurePetcareBridgeHandler(Bridge bridge, SurePetcareAPIHelper petcareAPI) {
+        super(bridge);
+        this.petcareAPI = petcareAPI;
+    }
+
+    @Override
+    @SuppressWarnings("null")
+    public void initialize() {
+        logger.debug("Initializing Sure Petcare bridge handler.");
+        SurePetcareBridgeConfiguration config = getConfigAs(SurePetcareBridgeConfiguration.class);
+        updateState(BRIDGE_CHANNEL_ONLINE, OnOffType.OFF);
+
+        if (config.getUsername() != null && config.getPassword() != null) {
+            updateStatus(ThingStatus.UNKNOWN);
+            try {
+                logger.debug("Login to SurePetcare API with username: {}", config.getUsername());
+                petcareAPI.login(config.getUsername(), config.getPassword());
+                logger.debug("Login successful, updating topology cache");
+                petcareAPI.updateTopologyCache();
+                logger.debug("Cache update successful, setting bridge status to ONLINE");
+                updateStatus(ThingStatus.ONLINE);
+                updateState(BRIDGE_CHANNEL_ONLINE, OnOffType.ON);
+                updateThings();
+            } catch (AuthenticationException e) {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+        } else {
+            logger.warn("Setting thing '{}' to OFFLINE: Parameter 'password' and 'username' must be configured.",
+                    getThing().getUID());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error-missing-username-or-password");
+        }
+
+        if (config.getRefreshIntervalTopology() != null) {
```

Comment:
Consider splitting these parts up in different methods, it seems you already divided it into parts by putting it in separate methods you can give it a name and it will be more readable and easier to maintain for others as well.